src/main/scala/chess/format/Forsyth.scala

```diff
@@ -108,25 +115,27 @@ object Forsyth {
     }
   }
 
-  case class SituationPlus(situation: Situation, fullTurnCount: Int) {
+  case class SituationPlus(situation: Situation, fullTurnCount: Int, isHalfTurn: Boolean = false) {
```

Comment:
So I would remove `isHalfTurn` and just access `situation.board.history.currentTurn.nonEmpty` instead to get the same info.